config/kernel-global_page_state.m4

```diff
@@ -0,0 +1,69 @@
+dnl #
+dnl # 4.8 API change
+dnl # kernel vm counters change
+dnl #
+AC_DEFUN([ZFS_AC_KERNEL_GLOBAL_NODE_PAGE_STATE], [
+	AC_MSG_CHECKING([whether global_none_page_state() exists])
+	ZFS_LINUX_TRY_COMPILE_SYMBOL([
+		#include <linux/mm.h>
+		#include <linux/vmstat.h>
+	],[
+		(void) global_node_page_state(0);
+	], [global_node_page_state], [include/linux/vmstat.h], [
+	],[
+		AC_MSG_RESULT(yes)
+		AC_DEFINE(ZFS_GLOBAL_NODE_PAGE_STATE, 1, [global_node_page_state() exists])
+	],[
+		AC_MSG_RESULT(no)
+	])
+])
+
+dnl #
+dnl # 4.14 API change (c41f012ade)
+dnl # mm: rename global_page_state to global_zone_page_state
+dnl #
+AC_DEFUN([ZFS_AC_KERNEL_GLOBAL_ZONE_PAGE_STATE], [
+	AC_MSG_CHECKING([whether global_zone_page_state() exists])
+	ZFS_LINUX_TRY_COMPILE_SYMBOL([
+		#include <linux/mm.h>
+		#include <linux/vmstat.h>
+	],[
+		(void) global_zone_page_state(0);
+	], [global_zone_page_state], [include/linux/vmstat.h], [
+		AC_MSG_RESULT(yes)
+		AC_DEFINE(ZFS_GLOBAL_ZONE_PAGE_STATE, 1, [global_zone_page_state() exists])
+	],[
+		AC_MSG_RESULT(no)
+	])
+])
+
+dnl #
+dnl # Create a define for an enum member
+dnl #
+AC_DEFUN([ZFS_AC_KERNEL_ENUM_MEMBER], [
+	AC_MSG_CHECKING([whether enum $2 contains $1])
+	AS_IF([AC_TRY_COMMAND("${srcdir}/scripts/enum-extract.pl" "$2" "$3" | egrep -qx $1)],[
```

Comment:
In general I'd prefer to avoid adding a helper script for this.  As you discovered already with the bots adding one further complicates a the build process.  It might be uglier, but doing the check inline here should  on the whole be simpler.  Also while I don't have anything against perl we should probably avoid it and somehow do this check with shell utilities.  Currently the build system doesn't depend on perl anywhere else and it would be nice to at least try and keep the dependencies as minimal as possible.

Reply:
`autogen.sh` calls `autoreconf` calls `autom4te`, and `autom4te` is a perl script - i.e. no additional dependency?

Now that I've gone through the (surprisingly tricky!) process of adding the helper script it might be worth leaving it there (and using it), as a kind of documentation?

Whilst I'm sure doing the check inline would be possible, I think it would be significantly harder to make it robust against things like multi-line comments and multi-line initialisers etc.

I really wouldn't be looking forward to trying it - how badly do you want to go inline with this?

Reply:
OK, you've talked me in to it.  Let's go with the helper script.